File: internal/model/resume_draft.go
package model

import "time"

type ResumeDraftRecord struct {
	ID        int64     `gorm:"primaryKey"`
	Name      string    `gorm:"column:name;type:TEXT;not null"`
	Note      string    `gorm:"column:note;type:TEXT;not null;default:''"`
	Data      []byte    `gorm:"column:data;type:TEXT;not null"`
	CreatedAt time.Time `gorm:"column:created_at;autoCreateTime"`
	UpdatedAt time.Time `gorm:"column:updated_at;autoUpdateTime"`
}

func (ResumeDraftRecord) TableName() string {
	return "resume_drafts"
}

File: web/src/types/resumeDraft.ts
import type { ResumeData } from "./resume";

export interface ResumeDraftSummary {
  id: number;
  name: string;
  note: string;
  createdAt: string;
  updatedAt: string;
}

export interface ResumeDraftDetail extends ResumeDraftSummary {
  data: ResumeData;
}

export interface CreateResumeDraftInput {
  name: string;
  note: string;
  data: ResumeData;
}
